node-notes-cli/app.js

```diff
@@ -0,0 +1,52 @@
+import { writeFile } from 'node:fs/promises';
+import { readFile } from 'node:fs/promises';
+
+if (process.argv[2] === "read") {
+  try {
+    const jsonData =  JSON.parse(await readFile("./data.json"));
+    for (let keys in jsonData.notes) {
+      console.log(`${keys}: ${jsonData.notes[keys]}`)
+    }
+  } catch (err) {
+    console.log(err);
+  }
+} else if (process.argv[2] === "create") {
+  try {
+    const jsonData = JSON.parse(await readFile("./data.json"));
+    const jsonID = JSON.stringify(jsonData.nextId, null, 2);
+    let noteArray = Object.entries(jsonData.notes).map(([k, v]) => ({ [k]: v }));
+    noteArray.push({[jsonID]: `${process.argv[3]}`});
+    let newObj = {}
+    for (let x = 0; x < noteArray.length; x++) {
+      Object.assign(newObj, noteArray[x]);
+    }
+    jsonData.notes = newObj
```

Comment:
Lines 17-23 are a very complex way to simply add an entry to an object. You should be able to do it in one (short) line of code. Do you remember the `[]` syntax for adding an entry to an object?